### PR TITLE
Added a scala testclass to test newmap files for lexing.

### PR DESCRIPTION
**Pull request by ajita-asthana**
Implemented automated lexer testing file for nm files. Next will try for parse tests.

### CODE REVIEW

File: src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      Identifier("myString"),
+      Symbol("="),
+      Identifier("hello"),
+      Number(67),
+      Identifier("world"),
+      Number(28)
+    )))
+  }
+
+  it should "divide up when starting with a 0" in {
```
Comment (max-winware) [inline src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala:46]: Should it?

I don't have any particular use case in mind there, but perhaps we could make leading zeros mean something

File: src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package ai.newmap.interpreter
+
+import ai.newmap.interpreter.Lexer.{DQuote, Enc, Identifier, Number, Symbol}
+import org.scalatest._
+import ai.newmap.model._
+import ai.newmap.util.{Failure, Success}
+import scala.io.Source
+
+
+class TestNewmapFiles extends FlatSpec{
```
Comment (max-winware) [inline src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala:10]: Are we only testing the lexer in this file? Maybe name it accordingly

**Review by max-winware — CHANGES_REQUESTED** [review]
I raised a couple of questions, but nearly good to go

**Review by maxsklar — CHANGES_REQUESTED** [review]
Woops I reviewed on the wrong account! Switching

File: src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala (hunk 2 above)
Comment (ajita-asthana) [reply src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala:10]: I was thinking of adding the parsing tests in the same file.

**Comment by maxsklar** [comment]
Ok - works for me then!

**Review by maxsklar — APPROVED** [review]

File: src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala (hunk 1 above)
Comment (ajita-asthana) [reply src/test/scala/ai/newmap/interpreter/TestNewmapFiles.scala:46]: Maybe leading zero in a number could signify as a negative number, should I remove this particular case for now?

**Comment by maxsklar** [comment]
We can keep it for now - easy enough to remove later.

It’s probably not a good idea to make it negative - I’m just thinking how
unintuitive that would be. Maybe we want a way to denote hex numbers or
something? That’s all optional at this point

**Pull request merged**